searchlib/src/vespa/searchlib/queryeval/blueprint.cpp

```diff
@@ -365,6 +370,16 @@ IntermediateBlueprint::optimize(Blueprint* &self)
     maybe_eliminate_self(self, get_replacement());
 }
 
+void
+IntermediateBlueprint::set_global_filter(std::shared_ptr<BitVector> global_filter)
```

Comment:
We should consider using a wrapper class (ex GlobalFilter) that owns the bitvector and implements shared_from_this to avoid over-copying the shared pointer.

Reply:
good catch, we'll do that soon.